src/field-builder.ts

```diff
@@ -63,7 +63,13 @@ fieldBuilderProto.effect = function effect({ effect = {}, resolve, ...options })
         }
       }
 
+      // fixme: result should not be Exit.Exit<never, any>
+      // because never cannot be checked if it's an instanceof Error
       if (Exit.isFailure(result)) {
+        if (Cause.isFailType(result.cause) && (result.cause.error as any instanceof Error)) {
+          throw result.cause.error;
+        }
```

Comment:
`result.cause.error` could can be not an instance of `Error`. so it should be wraped by `failErrorConstructor`, `effectOptions.defaultFailErrorConstructor` or `Error`.

Reply:
I might be misunderstanding you but that's exactly the reason for which I'm also checking if `result.cause.error instanceof Error` in the above code. If it isn't, I want it to fall through and be handled by the throw logic below:

```ts
throw new (effect.failErrorConstructor ?? effectOptions?.defaultFailErrorConstructor ?? Error)(
  Cause.pretty(result.cause),
);
```

Reply:
ahh yep. you're right.